studio/gcloud_artifact_store.py

```diff
@@ -16,42 +17,46 @@ def __init__(self, config, verbose=10, measure_timestamp_diff=True):
         self.logger = logging.getLogger('GCloudArtifactStore')
         self.logger.setLevel(verbose)
 
-        auth_config = config.get('auth')
-        if not auth_config:
-            self.client = storage.Client()
-        else:
-            assert auth_config['type'].lower() == 'firebase'
-            app = pyrebase.initialize_app(auth_config)
-            self.auth = FirebaseAuth(app,
-                                     auth_config.get("use_email_auth"),
-                                     auth_config.get("email"),
-                                     auth_config.get("password"))
-
-            self.client = storage.Client(credentials=self.auth.get_token())
+        self.config = config
 
         try:
-            self.bucket = self.client.get_bucket(config['bucket'])
+            self.bucket = self.getclient().get_bucket(config['bucket'])
         except BaseException as e:
             self.logger.exception(e)
-            self.bucket = self.client.create_bucket(config['bucket'])
+            self.bucket = self.getclient().create_bucket(config['bucket'])
 
         super(GCloudArtifactStore, self).__init__(measure_timestamp_diff)
 
+    def getclient(self):
```

Comment:
this is a tricky one - apparently, credentials that are created when one initializes storage.Client() expire after a while, so same instance of storage.Client() cannot be used for too long. Was causing problems within google app engine.